packages/advanced-logic/src/extensions/payment-network/any-to-erc20-proxy.ts

```diff
@@ -135,7 +132,7 @@ export default class AnyToErc20ProxyPaymentNetwork extends Erc20FeeProxyPaymentN
       return;
     }
 
-    if (!conversionSupportedNetworks.includes(network)) {
+    if (!erc20ConversionProxy.isDeployedOnNetwork(network)) {
```

Comment:
just to challenge: do we absolutely need this check? Or, do we need it to be done that "low" in the protocol? maybe the client could handle it

Reply:
I know that in the RF point of view, we could get rid of it since we make similar checks (several times, I think) before creating the request. (When we check for an existing conversion path, for example).

On the protocol level, I'm not sure this is checked elsewhere, and imo it wouldn't be a good thing for someone using the protocol as is, without performing checks on his side, to be able to create a request that cannot be paid...

Reply:
indeed! but the request-client could cover that. 

Other solution: do it in the `payment-detection` package. There's a long overdue refactoring to remove payment networks from advanced logic anyway...

Reply:
(the Factory is probably a good place)

Reply:
I'll look into it, I was not that happy to add a dependency either